diff -ruN qtbase-opensource-src-5.9.3-orig/mkspecs/common/gcc-base-mac.conf qtbase-opensource-src-5.9.3/mkspecs/common/gcc-base-mac.conf
--- qtbase-opensource-src-5.9.3-orig/mkspecs/common/gcc-base-mac.conf	2017-11-15 23:15:28.000000000 -0600
+++ qtbase-opensource-src-5.9.3/mkspecs/common/gcc-base-mac.conf	2018-01-12 20:44:35.000000000 -0600
@@ -15,7 +15,7 @@
 QMAKE_COMPILER_DEFINES += __APPLE__ __GNUC__=4 __APPLE_CC__
 
 QMAKE_LFLAGS_SHLIB         += -single_module -dynamiclib
-QMAKE_LFLAGS_PLUGIN        += $$QMAKE_LFLAGS_SHLIB
+QMAKE_LFLAGS_PLUGIN        += -bundle
 QMAKE_LFLAGS_INCREMENTAL   += -undefined suppress -flat_namespace
 QMAKE_LFLAGS_SONAME        += -install_name$${LITERAL_WHITESPACE}
 QMAKE_LFLAGS_HEADERPAD     += -headerpad_max_install_names
diff -ruN qtbase-opensource-src-5.9.3-orig/mkspecs/common/mac.conf qtbase-opensource-src-5.9.3/mkspecs/common/mac.conf
--- qtbase-opensource-src-5.9.3-orig/mkspecs/common/mac.conf	2017-11-15 23:15:28.000000000 -0600
+++ qtbase-opensource-src-5.9.3/mkspecs/common/mac.conf	2018-01-12 20:48:07.000000000 -0600
@@ -12,8 +12,8 @@
 
 include(unix.conf)
 
-QMAKE_RESOURCE          = /Developer/Tools/Rez
-QMAKE_EXTENSION_SHLIB   = dylib
+QMAKE_RESOURCE          = @XCODE_PREFIX@/Developer/Tools/Rez
+QMAKE_EXTENSION_SHLIB   = so
 QMAKE_LIBDIR            =
 
 # sdk.prf will prefix the proper SDK sysroot
diff -ruN qtbase-opensource-src-5.9.3-orig/mkspecs/features/create_cmake.prf qtbase-opensource-src-5.9.3/mkspecs/features/create_cmake.prf
--- qtbase-opensource-src-5.9.3-orig/mkspecs/features/create_cmake.prf	2017-11-15 23:15:28.000000000 -0600
+++ qtbase-opensource-src-5.9.3/mkspecs/features/create_cmake.prf	2018-01-12 20:48:24.000000000 -0600
@@ -147,7 +147,7 @@
         }
     } else {
         mac {
-            isEmpty(CMAKE_STATIC_TYPE): CMAKE_PlUGIN_EXT = .dylib
+            isEmpty(CMAKE_STATIC_TYPE): CMAKE_PlUGIN_EXT = .so
             else: CMAKE_PlUGIN_EXT = .a
 
             CMAKE_PLUGIN_LOCATION_RELEASE = $$PLUGIN_TYPE/lib$${TARGET}$${CMAKE_PlUGIN_EXT}
diff -ruN qtbase-opensource-src-5.9.3-orig/mkspecs/features/mac/rez.prf qtbase-opensource-src-5.9.3/mkspecs/features/mac/rez.prf
--- qtbase-opensource-src-5.9.3-orig/mkspecs/features/mac/rez.prf	2017-11-15 23:15:28.000000000 -0600
+++ qtbase-opensource-src-5.9.3/mkspecs/features/mac/rez.prf	2018-01-12 22:23:15.000000000 -0600
@@ -1,7 +1,7 @@
 
 #global defaults
 isEmpty(QMAKE_REZ) {
-    QMAKE_REZ = /Developer/Tools/Rez
+    QMAKE_REZ = @XCODE_PREFIX@/Developer/Tools/Rez
 }
 isEmpty(REZ_DIR):REZ_DIR = .
 isEmpty(QMAKE_EXT_REZ):QMAKE_EXT_REZ = .rsrc
diff -ruN qtbase-opensource-src-5.9.3-orig/mkspecs/features/qt_module.prf qtbase-opensource-src-5.9.3/mkspecs/features/qt_module.prf
--- qtbase-opensource-src-5.9.3-orig/mkspecs/features/qt_module.prf	2017-11-15 23:15:28.000000000 -0600
+++ qtbase-opensource-src-5.9.3/mkspecs/features/qt_module.prf	2018-01-12 20:49:44.000000000 -0600
@@ -264,7 +264,7 @@
 load(qt_targets)
 
 # this builds on top of qt_common
-!internal_module:!lib_bundle:if(unix|mingw) {
+unix|mingw {
     CONFIG += create_pc
     QMAKE_PKGCONFIG_DESTDIR = pkgconfig
     host_build: \
diff -ruN qtbase-opensource-src-5.9.3-orig/mkspecs/macx-clang/qmake.conf qtbase-opensource-src-5.9.3/mkspecs/macx-clang/qmake.conf
--- qtbase-opensource-src-5.9.3-orig/mkspecs/macx-clang/qmake.conf	2017-11-15 23:15:28.000000000 -0600
+++ qtbase-opensource-src-5.9.3/mkspecs/macx-clang/qmake.conf	2018-01-13 12:20:12.000000000 -0600
@@ -2,7 +2,7 @@
 # qmake configuration for Clang on OS X
 #
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.10
+QMAKE_MACOSX_DEPLOYMENT_TARGET = @MACOSX_DEPLOYMENT_TARGET@
 
 QMAKE_APPLE_DEVICE_ARCHS = x86_64
 
diff -ruN qtbase-opensource-src-5.9.3-orig/mkspecs/macx-g++/qmake.conf qtbase-opensource-src-5.9.3/mkspecs/macx-g++/qmake.conf
--- qtbase-opensource-src-5.9.3-orig/mkspecs/macx-g++/qmake.conf	2017-11-15 23:15:28.000000000 -0600
+++ qtbase-opensource-src-5.9.3/mkspecs/macx-g++/qmake.conf	2018-01-12 22:37:43.000000000 -0600
@@ -10,7 +10,7 @@
 CONFIG                 += app_bundle incremental global_init_link_order lib_version_first
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.10
+QMAKE_MACOSX_DEPLOYMENT_TARGET = @MACOSX_DEPLOYMENT_TARGET@
 
 QMAKE_APPLE_DEVICE_ARCHS = x86_64
 
diff -ruN qtbase-opensource-src-5.9.3-orig/mkspecs/macx-icc/qmake.conf qtbase-opensource-src-5.9.3/mkspecs/macx-icc/qmake.conf
--- qtbase-opensource-src-5.9.3-orig/mkspecs/macx-icc/qmake.conf	2017-11-15 23:15:28.000000000 -0600
+++ qtbase-opensource-src-5.9.3/mkspecs/macx-icc/qmake.conf	2018-01-12 22:38:59.000000000 -0600
@@ -95,7 +95,7 @@
 QMAKE_CFLAGS_HIDESYMS   += -fvisibility=hidden
 QMAKE_CXXFLAGS_HIDESYMS += $$QMAKE_CFLAGS_HIDESYMS -fvisibility-inlines-hidden
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.10
+QMAKE_MACOSX_DEPLOYMENT_TARGET = @MACOSX_DEPLOYMENT_TARGET@
 
 QMAKE_APPLE_DEVICE_ARCHS = x86_64
 
diff -ruN qtbase-opensource-src-5.9.3-orig/src/platformsupport/fontdatabases/mac/coretext.pri qtbase-opensource-src-5.9.3/src/platformsupport/fontdatabases/mac/coretext.pri
--- qtbase-opensource-src-5.9.3-orig/src/platformsupport/fontdatabases/mac/coretext.pri	2017-11-15 23:15:28.000000000 -0600
+++ qtbase-opensource-src-5.9.3/src/platformsupport/fontdatabases/mac/coretext.pri	2018-01-12 21:09:19.000000000 -0600
@@ -7,6 +7,12 @@
     SOURCES += freetype/qfontengine_ft.cpp
 }
 
+contains(QT_CONFIG, system-freetype) {
+    HEADERS += $$QT_SOURCE_TREE/src/gui/text/qfontengine_ft_p.h
+    SOURCES += $$QT_SOURCE_TREE/src/gui/text/qfontengine_ft.cpp
+    CONFIG += opentype
+}
+
 LIBS_PRIVATE += \
     -framework CoreFoundation \
     -framework CoreGraphics \
diff -ruN qtbase-opensource-src-5.9.3-orig/src/plugins/platforms/cocoa/cocoa.pro qtbase-opensource-src-5.9.3/src/plugins/platforms/cocoa/cocoa.pro
--- qtbase-opensource-src-5.9.3-orig/src/plugins/platforms/cocoa/cocoa.pro	2017-11-15 23:15:28.000000000 -0600
+++ qtbase-opensource-src-5.9.3/src/plugins/platforms/cocoa/cocoa.pro	2018-01-12 21:10:38.000000000 -0600
@@ -74,6 +74,10 @@
 
 LIBS += -framework AppKit -framework Carbon -framework IOKit -lcups
 
+contains(QT_CONFIG, system-freetype) {
+    LIBS += -lfreetype
+}
+
 QT += \
     core-private gui-private \
     accessibility_support-private clipboard_support-private theme_support-private \
